python-checks/src/main/java/org/sonar/python/checks/tests/UnconditionalAssertionCheck.java

```diff
@@ -0,0 +1,157 @@
+import org.sonar.python.tree.TreeUtils;
+
+import static org.sonar.plugins.python.api.tree.Tree.Kind.NAME;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.NUMERIC_LITERAL;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.QUALIFIED_EXPR;
+import static org.sonar.python.checks.CheckUtils.isClassOrFunction;
+import static org.sonar.python.checks.CheckUtils.isConstant;
+
+@Rule(key = "S5914")
+public class UnconditionalAssertionCheck extends PythonSubscriptionCheck {
+
+  private static final List<String> BOOLEAN_ASSERTIONS = List.of("assertTrue", "assertFalse");
+  private static final List<String> NONE_ASSERTIONS = List.of("assertIsNone", "assertIsNotNone");
+  private static final List<String> IS_ASSERTIONS = List.of("assertIs", "assertIsNot");
+
+  private static final String BOOLEAN_MESSAGE = "Replace this expression; its boolean value is constant.";
+  private static final String NONE_MESSAGE = "Remove this identity assertion; its value is constant.";
+  private static final String IS_MESSAGE = "Replace this \"assertIs\" call with an \"assertEqual\" call.";
+  private static final String IS_NOT_MESSAGE = "Replace this \"assertIsNot\" call with an \"assertNotEqual\" call.";
+  private static final String IS_SECONDARY_MESSAGE = "This expression creates a new object every time.";
+
+  private ReachingDefinitionsAnalysis reachingDefinitionsAnalysis;
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.FILE_INPUT, ctx ->
+      reachingDefinitionsAnalysis = new ReachingDefinitionsAnalysis(ctx.pythonFile()));
+
+    context.registerSyntaxNodeConsumer(Tree.Kind.ASSERT_STMT, ctx -> {
+      AssertStatement assertStatement = (AssertStatement) ctx.syntaxNode();
+      Expression condition = assertStatement.condition();
+      if (!condition.is(Tree.Kind.TUPLE) && !isAssertFalse(condition) && CheckUtils.isConstant(condition)) {
+        ctx.addIssue(condition, BOOLEAN_MESSAGE);
+      }
+    });
+
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, ctx -> {
+      CallExpression call = (CallExpression) ctx.syntaxNode();
+      Symbol symbol = call.calleeSymbol();
+      if (symbol == null) {
+        return;
+      }
+
+      String name = symbol.name();
+      List<Argument> arguments = call.arguments();
+
+      if (BOOLEAN_ASSERTIONS.contains(name)) {
+        checkBooleanAssertion(ctx, TreeUtils.nthArgumentOrKeyword(0, "testValue", arguments));
+      } else if (NONE_ASSERTIONS.contains(name)) {
+        checkNoneAssertion(ctx, call, TreeUtils.nthArgumentOrKeyword(0, "testValue", arguments));
+      } else if (IS_ASSERTIONS.contains(name)) {
+        String message = "assertIs".equals(name) ? IS_MESSAGE : IS_NOT_MESSAGE;
+        checkIsAssertion(ctx, call, TreeUtils.nthArgumentOrKeyword(0, "firstValue", arguments), message);
+        checkIsAssertion(ctx, call, TreeUtils.nthArgumentOrKeyword(1, "secondValue", arguments), message);
```

Comment:
Where did you see the keyword is `firstValue` and `secondValue`? In my `unittest.TestCase`, I have `expr1` and `expr2` as keywords (Python 3.10).